src/components/EstimateComboBox.tsx

```diff
@@ -139,53 +144,72 @@ export const EstimateComboBox = React.forwardRef<HTMLDivElement, EstimateComboBo
         }, [defaultValuePlaceholder, locale]);
 
         const onQButtonClick = useCallback(
-            (nextQ: string) => () => {
+            (nextQ: string | undefined) => () => {
                 setSelectedQ(nextQ);
-                setChanged(true);
+                if (nextQ === undefined) {
+                    setButtonText(currentYear);
+                } else {
+                    setChanged(true);
+                }
             },
-            [],
+            [currentYear],
         );
 
-        const onInputChange = useCallback((e: React.ChangeEvent<HTMLInputElement>) => {
-            setChanged(true);
-            setInputState(e.target.value);
+        const onInputYearChange = useCallback((e: React.ChangeEvent<HTMLInputElement>) => {
+            setCurrentYear(e.target.value);
         }, []);
 
-        useEffect(() => {
-            if (Object.keys(quarterInfo).length) {
-                setInputState(createLocaleDate(parseLocaleDate(quarterInfo[selectedQ].date, { locale }), { locale }));
-            }
-        }, [selectedQ, locale, quarterInfo]);
+        const onInputChange = useCallback(
+            (e: React.ChangeEvent<HTMLInputElement>) => {
+                setChanged(true);
+                setInputState(e.target.value);
+                if (isValidDate(e.target.value)) {
+                    setSelectedQ(quarterFromDate(parseLocaleDate(inputState, { locale })));
+                }
+            },
+            [inputState, locale],
+        );
 
         useEffect(() => {
-            if (isValidDate(inputState)) {
-                setSelectedQ(quarterFromDate(parseLocaleDate(inputState, { locale })));
+            if (Object.keys(quarterInfo).length && selectedQ !== undefined) {
```

Comment:
What the case to have 0 in `Object.keys(quarterInfo).length`?

Reply:
at the moment this case can no longer be, I will remove it